src/test/java/com/sivalabs/ft/features/api/controllers/FeatureDependencyEndpointsIntegrationTests.java

```diff
@@ -0,0 +1,598 @@
+    private JdbcTemplate jdbcTemplate;
+
+    @BeforeEach
+    void setUp() {
+        // Clean up any existing dependencies for test features
+        jdbcTemplate.update(
+                "DELETE FROM feature_dependencies WHERE feature_code IN ('IDEA-1', 'IDEA-2', 'GO-3', 'TEST-1', 'TEST-2')");
+    }
+
+    // === DEPENDENCIES ENDPOINT TESTS ===
+
+    @Test
+    @WithMockOAuth2User(username = "user")
+    void shouldGetDependenciesForFeature() {
+        // Given - Create some dependencies
+        createTestDependency("IDEA-1", "IDEA-2", "HARD", "Hard dependency");
+        createTestDependency("IDEA-1", "GO-3", "SOFT", "Soft dependency");
+
+        // When - Get dependencies
+        var result = mvc.get()
+                .uri("/api/features/{featureCode}/dependencies", "IDEA-1")
+                .exchange();
+
+        // Then
+        assertThat(result)
+                .hasStatusOk()
+                .hasContentType(MediaType.APPLICATION_JSON)
+                .bodyJson()
+                .extractingPath("$.size()")
+                .asNumber()
+                .isEqualTo(2);
+    }
+
+    @Test
+    @WithMockOAuth2User(username = "user")
+    void shouldReturnEmptyListWhenFeatureHasNoDependencies() {
+        // When - Get dependencies for feature with no dependencies
+        var result = mvc.get()
+                .uri("/api/features/{featureCode}/dependencies", "IDEA-2")
+                .exchange();
+
+        // Then
+        assertThat(result)
+                .hasStatusOk()
+                .bodyJson()
+                .extractingPath("$.size()")
+                .asNumber()
+                .isEqualTo(0);
+    }
+
+    @Test
+    @WithMockOAuth2User(username = "user")
+    void shouldReturn404WhenFeatureDoesNotExist() {
+        // When - Get dependencies for non-existent feature
+        var result = mvc.get()
+                .uri("/api/features/{featureCode}/dependencies", "NON-EXISTENT")
+                .exchange();
+
+        // Then
+        assertThat(result).hasStatus(HttpStatus.NOT_FOUND);
+    }
+
+    @Test
+    @WithMockOAuth2User(username = "user")
+    void shouldFilterDependenciesByProductCode() {
+        // Given - Create dependencies
+        createTestDependency("IDEA-1", "IDEA-2", "HARD", "Hard dependency");
+        createTestDependency("IDEA-1", "GO-3", "SOFT", "Soft dependency");
+
+        // When - Get dependencies filtered by product code
+        var result = mvc.get()
+                .uri("/api/features/{featureCode}/dependencies?productCode=goland", "IDEA-1")
+                .exchange();
+
+        // Then
+        assertThat(result)
+                .hasStatusOk()
+                .bodyJson()
+                .extractingPath("$.size()")
+                .asNumber()
+                .isEqualTo(1);
+
+        // Verify the returned dependency
+        assertThat(result)
+                .bodyJson()
+                .extractingPath("$[0]")
+                .convertTo(FeatureDependencyDto.class)
```

Comment:
May be we should not use new dto in tests, though here it's pretty obvious that this Dto should work.

We for example could change methods to return Feature? Then we can use existing feature dto.